lib/castle/hourly_download.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Castle.HourlyDownload do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @primary_key false
+
+  schema "hourly_downloads" do
+    field :podcast_id, :integer
+    field :episode_id, :binary_id
+    field :dtim, :utc_datetime
+    field :count, :integer
+  end
+
+  @doc false
+  def changeset(download, attrs) do
+    download
+    |> cast(attrs, [:podcast_id, :episode_id, :dtim, :count])
+    |> validate_required([:podcast_id, :episode_id, :dtim, :count])
+  end
+
+  def upsert_all([]), do: 0
+  def upsert_all(rows) when length(rows) > 5000 do
+    Enum.chunk_every(rows, 5000)
+    |> Enum.map(&upsert_all/1)
+    |> Enum.sum()
+  end
+  def upsert_all(rows) do
+    Castle.Repo.insert_all Castle.HourlyDownload, Enum.map(rows, &parse_row/1)
```

Comment:
Because of the trigger/function on inserts for `castle_downloads` ... they actually function as upserts.